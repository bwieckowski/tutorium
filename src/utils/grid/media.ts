export type  BrakepointsName = 'mobile' | 'tablet' | 'ipad' | 'desktop' | 'lgDesktop';
type Brakepoints = {
  [name in BrakepointsName]: number
}

export const breakposints: Brakepoints  = {
  mobile : 320,
  tablet: 768,
  ipad: 992,
  desktop: 1200,
  lgDesktop: 2500
}


export const brakepointsNames = ['mobile','tablet' , 'ipad' , 'desktop' , 'lgDesktop'];
export const from = ( breakpoint: BrakepointsName ) =>( `@media(min-width: ${breakposints[breakpoint]}px)`);

