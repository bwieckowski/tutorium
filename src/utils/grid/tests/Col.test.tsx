import * as React from "react";
import * as renderer from "react-test-renderer"
import ColWrapper from "../Col"
import {BrakepointsName, breakposints} from "../media";
import 'jest-styled-components'

const brakepointsValue = Object.keys(breakposints).map( (key: BrakepointsName) =>(`(max-width: ${breakposints[key]}px)`));

it("renders the heading", () => {
    const result = renderer.create(<ColWrapper sizes={[1, 2, 1, 3,1]} />).toJSON();
    expect(result).toMatchSnapshot();
    expect(result).toHaveStyleRule('width', '100%', {
        media: brakepointsValue[0]
    });
});
