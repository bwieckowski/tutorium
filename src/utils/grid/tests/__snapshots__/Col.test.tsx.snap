// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (max-width:320px) {
  .c0 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c0 {
    width: 50%;
  }
}

@media (max-width:992px) {
  .c0 {
    width: 100%;
  }
}

@media (max-width:1200px) {
  .c0 {
    width: 33.333333333333336%;
  }
}

@media (max-width:2500px) {
  .c0 {
    width: 100%;
  }
}

<div
  className="c0"
  sizes={
    Array [
      1,
      2,
      1,
      3,
      1,
    ]
  }
/>
`;
